specs/agents/tracing-sampling.md

```diff
@@ -1,12 +1,97 @@
 #### Transaction sampling
 
-To reduce processing and storage overhead, transactions may be "sampled". Currently sampling has the effect of limiting the amount of data we capture for transactions: for non-sampled transactions instrumentation should not record context, nor should any spans be captured. The default graphs in the APM UI will utilise the transaction properties available for both sampled and non-sampled transactions.
+To reduce processing and storage overhead, transactions may be sampled by agents.
+Sampling here refers to "head-based sampling".
 
-By default all transactions will be sampled. Agents can be configured to sample probabilistically, by specifying a sampling
-probability in the range \[0,1\] using the configuration `ELASTIC_APM_TRANSACTION_SAMPLE_RATE`. For example:
+Head-based sampling is where a sampling decision is made at the root of the distributed trace,
+before the details or outcome of the trace are known,
+and propagated throughout the trace.
+
+##### Configuration
+
+By default all transactions will be sampled.
+Agents can be configured to sample probabilistically,
+by specifying a sampling probability in the range \[0,1\] using the configuration `ELASTIC_APM_TRANSACTION_SAMPLE_RATE`.
+e.g.
 
- - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0` means no transactions will be sampled
  - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=1` means all transactions will be sampled (the default)
+ - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0` means no transactions will be sampled
  - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0.5` means approximately 50% of transactions will be sampled
 
-If a transaction is not sampled, you should set the `sampled: false` property and omit collecting `spans` and `context`.
+##### Effect on metrics
+
+At the time of making a sampling decision,
+the sampling rate must be recorded so that it can be associated with every transaction and span in the trace.
+The sampling rate will be used by the server for scaling transaction and span metrics.
+
+Transaction metrics will be used by the UI to display transaction distributions and throughput,
+from the perspective of the transaction's service (grouped by `service.name` and `transaction.name`).
```

Comment:
@axw Probably no need to add this to the agents spec, more out of curiosity, what other dimensions are you planning to add? `outcome`? `result`? What's the cardinality limit for `transaction.name`? Are you planning to also track metrics without the `transaction.name` dimension to speed up queries that go across all transaction names? Could also be a special-case name like  `transaction.name: "_all"`.

Reply:
What are the current plans regarding the detection of dynamic parts in the URL?

Reply:
How important is it that all agents report `${request.method} unknown route` instead of `${request.method} ${request.path}` for the metrics collection to work properly? Is the metrics collection also scheduled for 7.10?

Reply:
> @axw Probably no need to add this to the agents spec, more out of curiosity, what other dimensions are you planning to add? outcome? result?

`result` is already in there, and `outcome` will be added once we've finalised the spec. The full list is at https://github.com/elastic/apm-server/blob/5cb4101d705effdf2f54e2e45847ee92033e806e/x-pack/apm-server/aggregation/txmetrics/aggregator.go#L414

> What's the cardinality limit for transaction.name?

The server has a configurable limit for the number of transaction metric buckets, which resets after every reporting interval. It's currently defaulting to 1000. Once that limit is reached, the server will start recording single-value metrics for new buckets.

> Are you planning to also track metrics without the transaction.name dimension to speed up queries that go across all transaction names? Could also be a special-case name like transaction.name: "_all".

Not currently planning to do this in the server, but the UI might create transforms based off the transaction group metrics: https://github.com/elastic/kibana/issues/74498 

> What are the current plans regarding the detection of dynamic parts in the URL?

None that I'm aware of. RUM recently introduced a heuristic-based approach: https://github.com/elastic/apm-agent-rum-js/pull/827

> How important is it that all agents report ${request.method} unknown route instead of ${request.method} ${request.path} for the metrics collection to work properly? Is the metrics collection also scheduled for 7.10?

Very important. If transaction names have such high cardinality, then distributions become less meaningful. I believe we're intending to preview the Metrics-based UI in 7.10.

Reply:
I've added some remarks to the Kibana issue: https://github.com/elastic/kibana/issues/74498#issuecomment-675291378

> Very important. If transaction names have such high cardinality, then distributions become less meaningful. I believe we're intending to preview the Metrics-based UI in 7.10.

That's good info, I'll work with the agent teams to ensure we're aligned.